parser/proto.go

```diff
@@ -0,0 +1,471 @@
+package parser
+
+import (
+	"encoding/binary"
+	"fmt"
+	"github.com/confluentinc/confluent-kafka-go/schemaregistry"
+	"github.com/confluentinc/confluent-kafka-go/schemaregistry/serde"
+	"github.com/housepower/clickhouse_sinker/model"
+	"github.com/housepower/clickhouse_sinker/util"
+	"github.com/jhump/protoreflect/desc"
+	"github.com/jhump/protoreflect/desc/protoparse"
+	"github.com/jhump/protoreflect/dynamic"
+	"github.com/shopspring/decimal"
+	"github.com/thanos-io/thanos/pkg/errors"
+	"golang.org/x/exp/constraints"
+	"google.golang.org/protobuf/types/known/timestamppb"
+	"io"
+	"math"
+	"reflect"
+	"regexp"
+	"strings"
+	"sync"
+	"time"
+)
+
+//go:generate mockgen -source=../vendor/github.com/confluentinc/confluent-kafka-go/schemaregistry/schemaregistry_client.go -imports=schemaregistry=github.com/confluentinc/confluent-kafka-go/schemaregistry -package=parser -mock_names=Client=MockSchemaRegistryClient -destination=schema_registry_mock_test.go
+
+// ProtoParser knows how to get data from proto format.
+type ProtoParser struct {
+	pp           *Pool
+	deserializer *ProtoDeserializer
+}
+
+// Parse parses bytes into metric that knows how.
+func (p *ProtoParser) Parse(bs []byte) (metric model.Metric, err error) {
```

Comment:
how about getting rid of named returned values? here and elsewhere in this file

Reply:
Yes, fixed this file